src/app/common/hooks/balance/stx/use-stx-available-balance.ts

```diff
@@ -0,0 +1,23 @@
+import { useMemo } from 'react';
+
+import { createMoney } from '@shared/models/money.model';
+import { isDefined } from '@shared/utils';
+
+import { subtractMoney } from '@app/common/money/calculate-money';
+import { useCurrentStacksAccountAnchoredBalances } from '@app/query/stacks/balance/stx-balance.hooks';
+
+import { useStxOutboundValue } from './use-stx-outbound-value';
+
+export function useStxAvailableBalance() {
+  const { data: balances } = useCurrentStacksAccountAnchoredBalances();
+  const unlockedStxBalance = balances?.stx.unlockedStx;
+  const stxOutboundValue = useStxOutboundValue();
+
+  return useMemo(
+    () =>
+      isDefined(unlockedStxBalance)
+        ? subtractMoney(unlockedStxBalance, stxOutboundValue)
+        : createMoney(0, 'STX'),
+    [unlockedStxBalance, stxOutboundValue]
+  );
```

Comment:
How does loading work? Does it flash from 0 to balance? Wonder if we should also use select here? What do you think?

Reply:
I restructured all this hooks to return loading state